lib/index.js

```diff
@@ -290,7 +326,7 @@ util.inherits(Device, EventEmitter);
  */
 function createSocket (type, opts) {
     var domain = (opts || {}).raw ? nn.AF_SP_RAW : nn.AF_SP;
-    return new Socket(domain, type);
+    return new Socket(domain, type, opts);
```

Comment:
delete `var domain ...` and the first arg.  No need to pull it out of `opts` here, since we pass on `opts`.  Fix up the `Socket` constructor too to pull the domain out of `opts`.